### DISCUSSION TITLE
Guidance for rolling back Dagster version if needed

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17821] [Q&A] [answered]
hello friends, apologies if this has been asked before and/or this is not the right place for this question.

What is the recommended way to do a down migration? I accidentally migrated my db to 1.5.6 when I meant to migrate just to 1.3.7.
The specific version is not a concern right now (we were planning to go to 1.5.6 anyway) - but it raised the question: what if there is a breaking change and we need to roll back dagster version, how do we do the down migration?

thanks!

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699459420946269?thread_ts=1699459420.946269&cid=C01U954MEER).*

**Comment by gibsondan** [comment, marked as answer]
Our recommendation for this is to back up the database before performing the migration. There's a recommended set of steps to follow here: https://docs.dagster.io/deployment/guides/kubernetes/how-to-migrate-your-instance#walkthrough - those are kubernetes specific but the underlying operations would work outside of Kubernetes too.
